### PR TITLE
Modify rule S5720: LaYC format

### PR DESCRIPTION
**Pull request by joke1196**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

**Review by maksim-grebeniuk-sonarsource — APPROVED** [review]
LGTM

File: rules/S5720/python/rule.adoc [hunk 1]
```diff
@@ -1,35 +1,60 @@
...
 
+This rule will accept "cls" or "mcs" as first parameter's name for class and metaclasses methods.
 
-This rule raises an issue when the first parameter of an instance method is not called "self".
+No issue will be raised for the following methods: ``++__init_subclass__++``, ``++__class_getitem__++`` and ``++__new__++``, as these methods' first parameter is a class.
 
+You can also disable raising issues on methods decorated with a specific decorators. These decorators can be added to this rule's "ignoredDecorators" parameter.
 
-=== Noncompliant code example
+For example, with "ignoredDecorators" set to "abstractmethod".
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S5720/python/rule.adoc:20]: Does it make sense to use `abstractmethod` here as an example?

As far as I know, the parameter is mostly there for custom decorators we may not have knowledge about and which might make the rule irrelevant. I don't think that's the case with `abstractmethod` though?

Maybe we could use a made-up name that makes things blurrier and understandable why we wouldn't want to raise (like custom static method or something)?

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Looks good, minor comments.

File: rules/S5720/python/rule.adoc [hunk 2]
```diff
@@ -1,35 +1,60 @@
...
 
-Note also that creating methods which are used as static methods without the ``++@staticmethod++`` decorator is a bad practice because calling these methods on an instance will raise a ``++TypeError++``. Either move the method out of the class or decorate it with ``++@staticmethod++``.
+=== Exceptions
 
+This rule will accept "cls" or "mcs" as first parameter's name for class and metaclasses methods.
 
-This rule raises an issue when the first parameter of an instance method is not called "self".
+No issue will be raised for the following methods: ``++__init_subclass__++``, ``++__class_getitem__++`` and ``++__new__++``, as these methods' first parameter is a class.
 
+You can also disable raising issues on methods decorated with a specific decorators. These decorators can be added to this rule's "ignoredDecorators" parameter.
```
[the remark this answers is not in the document]
Comment (joke1196) [reply rules/S5720/python/rule.adoc:18]: Yes sure I think it is clearer with the `prevent raising` bit. Thanks

File: rules/S5720/python/rule.adoc (hunk 1 above)
Comment (joke1196) [reply rules/S5720/python/rule.adoc:20]: Yes, you are right, it makes sense to use a custom decorator.

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM

File: rules/S5720/python/rule.adoc [hunk 3]
```diff
@@ -1,35 +1,58 @@
+This rule raises an issue when the first parameter of an instance method is not called "self".
+
 == Why is this an issue?
 
 Instance methods, i.e. methods not annotated with ``++@classmethod++`` or ``++@staticmethod++``, are expected to have at least one parameter. This parameter will reference the object instance on which the method is called. By convention, this first parameter is named "self".
 
+Naming the first parameter something different than "self" is not recommended as it could lead to confusion. It might indicate that the "self" parameter was forgotten, in which case calling the method will most probably fail.
```
Comment (jean-jimbo-sonarsource) [inline rules/S5720/python/rule.adoc:7]: Please mod to "...something different from "self"..."

File: rules/S5720/python/rule.adoc [hunk 4]
```diff
@@ -1,35 +1,58 @@
...
 
-Note also that creating methods which are used as static methods without the ``++@staticmethod++`` decorator is a bad practice because calling these methods on an instance will raise a ``++TypeError++``. Either move the method out of the class or decorate it with ``++@staticmethod++``.
+=== Exceptions
 
+This rule will accept "cls" or "mcs" as first parameter's name for class and metaclasses methods.
 
-This rule raises an issue when the first parameter of an instance method is not called "self".
+No issue will be raised for the following methods: ``++__init_subclass__++``, ``++__class_getitem__++`` and ``++__new__++``, as these methods' first parameter is a class.
 
+This rule may be parametrized to prevent raising issues on methods decorated with specific decorators. These decorators can be added to this rule's `ignoredDecorators` parameter.
```
Comment (jean-jimbo-sonarsource) [inline rules/S5720/python/rule.adoc:18]: spelling: "parameterized"

File: rules/S5720/python/rule.adoc [hunk 5]
```diff
@@ -1,35 +1,58 @@
...
+class MyClass(ABC):
+    @myDecorator
+    def method(arg):  # No issue will be raised.
+        pass
+----
+
+
+== How to fix it
+
+Make sure to have a "self" parameter on instance methods and annotate static method with the `@staticmethod` decorator.
```
Comment (jean-jimbo-sonarsource) [inline rules/S5720/python/rule.adoc:33]: "...and annotate static methods..."

**Review by jean-jimbo-sonarsource — COMMENTED** [review]
Left a few spelling/grammar comments

**Review by jean-jimbo-sonarsource — APPROVED** [review]

**Pull request merged**